src/components/ClaimMasterPanel.js

```diff
@@ -308,14 +308,14 @@ class ClaimMasterPanel extends FormPanel {
                 value={(edited.visitType === this.claimTypeReferSymbol ? edited.referFrom: edited.referTo) ?? this.EMPTY_STRING}
                 reset={reset}
                 readOnly={ro}
-                required={edited.visitType === this.claimTypeReferSymbol}
+                required={!!this.isReferHFMandatory&&(edited.visitType === this.claimTypeReferSymbol)}
```

Comment:
No need to '!!' this variable. It is boolean by default. Moreover add extra spacing to '&&'.